### ISSUE TITLE
Don't bind mariadb to localhost in a hidden place

### ISSUE THREAD

**Issue by elmar-hinz** [Homebrew/homebrew-core#12744]
# Formula: 

https://github.com/Homebrew/homebrew-core/blob/master/Formula/mariadb.rb

# Error

Error message: `Can't connect to MySQL server on '192.168.2.130' (61)`

# Issue:

The server is forced to bind to 127.0.0.1 in a rather hidden place. This is quite difficult to debug. 

Inside a plist file:   `<string>--bind-address=127.0.0.1</string>`

# Rational

Most advice found by google hint to `etc/my.conf` but not to the plist file.

Binding to 127.0.0.1 is only possible from localhost. This is good for security reasons. In times of virtual hosts and docker containers it is often necessary to be able to bind to the hosts real IP. It must be easy to enable that setting.

# Solution

Put the setting into `etc/my.conf`

**Comment by woodruffw** [comment]
See https://github.com/Homebrew/legacy-homebrew/issues/34629 and https://github.com/Homebrew/legacy-homebrew/pull/20090

This seems to be the solution we've settled on for both MariaDB and MySQL, since it involves a minimum amount of configuration wrangling on our part and is more secure by default.

**Comment by elmar-hinz** [comment]
For the average user it's to hard to debug. The full-blown admin will be able to debug it, but that's not the main target group of homebrew.

After all it breaks the functionality of the the actual configuration file, something I think wrong by conception. It seems contradicting the spirit of homebrew, not to put a redundant layer of configuration in front of the actual configuration files.

**Comment by woodruffw** [comment]
As far as I know, we don't target either group particularly.

Our primary mission (if we have one?) is to provide a a nice environment for developers on OS X. Part of that is providing safe defaults, and doing so in a way that isn't flaky (using regular expressions to rewrite a configuration file when we don't have to) or adds an additional maintenance burden (bundling our own default configuration file that just as many people will find doesn't suit them and will break if MariaDB/MySQL ever change their formats).

Since the vast majority of users haven't had (or at least reported) an issue with this, I'm inclined to say that things should stay the way they are (with the added benefit of being consistent with `mysql`, `hudson`, `jenkins`, etc). Of course, if people do continue to report this, changing it would probably be for the better.

**Comment by npinchot** [comment]
I spent too much time yesterday troubleshooting why only localhost could connect :( Another developer had the same issue and gave up with Homebrew MySQL and installed the download from their site. After a couple cups of ☕️ this morning and going through uninstalling/reinstalling MySQL formula I found the plist relatively quickly.

I don't want to reopen the debate about what the default should be. I think the easiest way to address this problem is to give some detail in the caveat section about the plist bind address hardcoding. I've opened PR #12948 for this.
